feedback_prompt_learning/search_algo/world_model.py

```
@@ -0,0 +1,399 @@
+            return []
+        batch_evals = all_evaluations[-num_examples:]
+
+        return batch_evals  # Return recent examples (could filter by score if needed)
+
+
+class PromptAgentWorldModel(PromptOptimizationWorldModel):
+    """
+    World model for prompt agents that do not require feedback-based sampling.
+    Uses random sampling from all evaluations.
+    """
+    def get_fixed_eval_batch(self, depth):
```

Ruff (0.14.8)

Unused method argument: `depth`

(ARG002)